conans/test/integration/workspace_test.py

```diff
@@ -249,81 +322,159 @@ def files(name, depend=None):
+        client.runner('cmake .. -G "%s" -DCMAKE_BUILD_TYPE=Release' % generator, cwd=build)
+        client.runner('cmake --build . --config Release', cwd=build)
+
+        cmd_release = os.path.normpath("./A/build/Release/app")
+        cmd_debug = os.path.normpath("./A/build/Debug/app")
+
         client.runner(cmd_release, cwd=client.current_folder)
         self.assertIn("Hello World C Release!", client.out)
         self.assertIn("Hello World B Release!", client.out)
         self.assertIn("Hello World A Release!", client.out)
 
-        # Now do the same for debug
-        client.run("install . -s build_type=Debug")
-        client.run("build C -bf=C/%s" % debug)
-        client.run("build B -bf=B/%s" % debug)
-        client.run("build A -bf=A/%s" % debug)
+        tools.replace_in_file(os.path.join(client.current_folder, "C/src/hello.cpp"),
+                              "Hello World", "Bye Moon", output=client.out)
+
+        client.runner('cmake --build . --config Release', cwd=build)
+        client.runner(cmd_release, cwd=client.current_folder)
+        self.assertIn("Bye Moon C Release!", client.out)
+        self.assertIn("Hello World B Release!", client.out)
+        self.assertIn("Hello World A Release!", client.out)
+
+        tools.replace_in_file(os.path.join(client.current_folder, "B/src/hello.cpp"),
+                              "Hello World", "Bye Moon", output=client.out)
+
+        client.runner('cmake --build . --config Release', cwd=build)
+        client.runner(cmd_release, cwd=client.current_folder)
+        self.assertIn("Bye Moon C Release!", client.out)
+        self.assertIn("Bye Moon B Release!", client.out)
+        self.assertIn("Hello World A Release!", client.out)
+
+        self.assertNotIn("Debug", client.out)
+
+        client.runner('cmake .. -G "%s" -DCMAKE_BUILD_TYPE=Debug' % generator, cwd=build)
+        # CMake configure will find the Release libraries, as we are in cmake-multi mode
+        # Need to reset the output after that
+        client.init_dynamic_vars()  # Reset output
+        client.runner('cmake --build . --config Debug', cwd=build)
+        client.runner(cmd_debug, cwd=client.current_folder)
+        self.assertIn("Bye Moon C Debug!", client.out)
+        self.assertIn("Bye Moon B Debug!", client.out)
+        self.assertIn("Hello World A Debug!", client.out)
+
+        tools.replace_in_file(os.path.join(client.current_folder, "C/src/hello.cpp"),
+                              "Bye Moon", "Hello World", output=client.out)
+
+        client.runner('cmake --build . --config Debug', cwd=build)
         client.runner(cmd_debug, cwd=client.current_folder)
         self.assertIn("Hello World C Debug!", client.out)
-        self.assertIn("Hello World B Debug!", client.out)
+        self.assertIn("Bye Moon B Debug!", client.out)
         self.assertIn("Hello World A Debug!", client.out)
+
+        self.assertNotIn("Release", client.out)
+
+    def build_requires_test(self):
```

Comment:
I thought this test was going to use a `build_requires` in editable mode. Does it worth another test?

Reply:
Yes, added.